src/resource/third.js

```diff
@@ -20,21 +20,56 @@ third.GetPath = function(path) {
  * @returns {HTMLScriptElement}
  */
 third.InjectScript = function(path, options) {
+  // note that options shouldn't have a "name" property, to keep the code for third.InjectToggleableScripts simpler.
   options = options || {}
   const el = document.createElement("script");
   el.src = third.GetPath(path);
   el.type = "text/javascript";
   el.async = options.noAsync ? false : true;
+  // Why not always use head?
   (options.useHead ? document.head : document.body).append(el);
   return el;
 }
 
 /**
- * Gets the default settings.
+ * Calls third.InjectScript on multiple scripts in a batch, only when the corresponding configuration options are truthy.
+ * @param {object} scripts An Object where keys are setting names and values are either injectable script names, objects with options (script name in "name" key), or arrays of these.
+ * @param {string} caller The name of the content script calling the function
+ */
+third.InjectToggleableScripts = async function(scripts, caller) {
+  const settings = await third.GetSettings();
+  const scriptsToInject = Object.entries(scripts).filter((entry) => settings[entry[0]]).map((entry) => entry[1]);
+  if (scriptsToInject.length == 0) { return; }
+  // TODO: Inject polyfilled utilities relevant to the content script
+  for (const script of scriptsToInject) {
+    if (typeof script === "string") {
+      third.InjectScript(script);
+    } else if (Array.isArray(script)) {
+      for (const trueScript of script) {
+        if (typeof trueScript === "string") {
+          third.InjectScript(trueScript);
+        } else if (typeof trueScript === "object") {
+          third.InjectScript(trueScript.name, trueScript);
+        } else {
+          // TODO: Throw error, see below
+        }
+      }
+    } else if (typeof script === "object") {
+      third.InjectScript(script.name, script);
+    } else {
+      // TODO: Throw error? idk how js handles that 
```

Comment:
```suggestion
      throw new Error("Script isn't of a valid type");
```

Reply:
Alternatively:
```suggestion
      console.error("Script isn't of a valid type");
```
This does the same thing, but without stopping the currently running script.